src/app/portlets/dot-edit-content/services/dot-edit-content-html.service.ts

```diff
@@ -77,43 +112,52 @@ export class DotEditContentHtmlService {
         this.dotContainerContentletService
             .getContentletToContainer(this.addContentContainerIdentifier, contentlet.identifier)
             .subscribe((contentletHtml: string) => {
-                const contentletContentEl = contentletEl.querySelector('.dotedit-contentlet__content');
-
-                // Removing the loading indicator
-                contentletContentEl.innerHTML = '';
-                this.appendNewContentlets(contentletContentEl, contentletHtml);
-
+                this.renderHTMLToContentlet(contentletEl, contentletHtml);
                 this.addContentContainerIdentifier = null;
-
-                // Update the model with the recently added contentlet
-                this.pageModelChange.next(this.getContentModel());
             });
     }
 
+    /**
+     * Set the container id where a contentlet will be added
+     *
+     * @param {string} identifier
+     * @memberof DotEditContentHtmlService
+     */
     setContainterToAppendContentlet(identifier: string): void {
         this.addContentContainerIdentifier = identifier;
     }
 
+    /**
+     * Return the page model
+     *
+     * @returns {*}
+     * @memberof DotEditContentHtmlService
+     */
     getContentModel(): any {
         return this.getEditPageIframe().contentWindow.getDotNgModel();
     }
 
     private addContentToolBars(): void {
         const doc = this.getEditPageDocument();
-        this.dotEditContentToolbarHtmlService.addContainerToolbar(doc).then(() => {
-            this.bindContainersEvents();
-        }).catch(error => {
-            this.loggerService.debug(error);
-        });
+        this.dotEditContentToolbarHtmlService
+            .addContainerToolbar(doc)
+            .then(() => {
+                this.bindContainersEvents();
+            })
+            .catch((error) => {
+                this.loggerService.debug(error);
+            });
 
-        this.dotEditContentToolbarHtmlService.addContentletMarkup(doc).then(() => {
-            this.bindContenletsEvents();
-        }).catch(error => {
-            this.loggerService.debug(error);
-        });
+        this.dotEditContentToolbarHtmlService
+            .addContentletMarkup(doc)
+            .then(() => {
+                this.bindContenletsEvents();
+            })
+            .catch((error) => {
+                this.loggerService.debug(error);
```

Comment:
Should we use the `dot-global-message.service.ts` here to display a possible error ?

Reply:
I don't know how are we going to handle this type of errors yet... need to figure it out.